test/dartdevc/dartdevc_test.dart

```diff
@@ -1,15 +1,20 @@
 // Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
+import 'dart:async';
+import 'dart:io';
 
+import 'package:path/path.dart' as p;
 import 'package:test/test.dart';
 
 import '../descriptor.dart' as d;
 import '../serve/utils.dart';
 import '../test_pub.dart';
 
 main() {
-  test("can compile js files for modules under lib and web", () async {
+  test(
+      "can compile js files for modules under lib and web, and handle new "
+      "files and edits.", () async {
```

Comment:
Created a group and moved shared setup code into a setUp to reduce the duplicate code and get the clean separation of tests.